src/webview/web_context.rs

```diff
@@ -0,0 +1,171 @@
+use std::path::{Path, PathBuf};
+
+/// A context that is shared between multiple [`WebView`]s.
+///
+/// A browser would have a context for all the normal tabs and a different context for all the
+/// private/incognito tabs.
+pub struct WebContext {
+  data: WebContextData,
+  #[allow(dead_code)]
+  os: WebContextImpl,
+}
+
+impl WebContext {
+  /// Create a new [`WebContext`].
+  ///
+  /// `data_directory`:
+  /// * Whether the WebView window should have a custom user data path. This is useful in Windows
+  ///   when a bundled application can't have the webview data inside `Program Files`.
+  pub fn new(data_directory: Option<PathBuf>) -> Self {
+    let data = WebContextData { data_directory };
+    let os = WebContextImpl::new(&data);
+    Self { data, os }
+  }
+
+  /// A reference to the data directory the context was created with.
+  pub fn data_directory(&self) -> Option<&Path> {
+    self.data.data_directory()
+  }
+}
+
+impl Default for WebContext {
+  fn default() -> Self {
+    let data = WebContextData::default();
+    let os = WebContextImpl::new(&data);
+    Self { data, os }
+  }
+}
+
+/// Data that all [`WebContext`] share regardless of platform.
+#[derive(Default)]
+struct WebContextData {
+  data_directory: Option<PathBuf>,
+}
+
+impl WebContextData {
+  /// A reference to the data directory the context was created with.
+  pub fn data_directory(&self) -> Option<&Path> {
+    self.data_directory.as_deref()
+  }
+}
+
+#[cfg(not(target_os = "linux"))]
+#[cfg_attr(doc_cfg, doc(cfg(not(target_os = "linux"))))]
```

Comment:
We don't need doc_cfg, the doc should create the right one according to `#[cfg(not(target_os = "linux"))]`

Reply:
oops that was leftover from when i merged the windows and mac one into a single unit struct